### PR TITLE
Request IR by text message

### PR DESCRIPTION
**Pull request by iwootten**
### What is the context of this PR?

Adds the journey to request an individual response by text message shown in the prototype. Encrypts the mobile number from view in the url. Validation is not handled and will follow on a separate card.

NB: I couldn't see why the person_id would be present in the url so I've not included it. This also follows how the post url structure. I'm away until the end of the month so this will need to be picked up by someone else in my absence.

### How to review 

See if you agree with the approach.

### CODE REVIEW

File: tests/integration/individual_response/test_individual_response.py [hunk 1]
```diff
@@ -433,6 +433,114 @@ def test_previous_from_confirm_returns_via_hub_route(self):
         self.assertInUrl(f"/individual-response/{list_item_id}/how?journey=hub")
 
 
+class TestIndividualResponseTextHandler(IndividualResponseTestCase):
+    def test_display_mobile_on_confirmation_page(self):
```
Comment (JamesGardiner) [inline tests/integration/individual_response/test_individual_response.py:437]:
```suggestion
    def test_display_mobile_number_on_confirmation_page(self):
```

File: templates/individual_response/mobile_confirmation.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{% extends 'layouts/_base.html' %}
```
Comment (JamesGardiner) [inline templates/individual_response/mobile_confirmation.html:1]: I don't think mobile_confirmation is the best name for this template. `sms_confirmation` or `text_message_confirmation` would be more explicit

File: tests/integration/individual_response/test_individual_response.py [hunk 3]
```diff
@@ -433,6 +433,114 @@ def test_previous_from_confirm_returns_via_hub_route(self):
...
+        # Given I navigate to the confirmation page
+        self._add_household_no_primary()
+        self.get(self.individual_section_link)
+        self.get(self.individual_response_link)
+        self.post()
+        self.post({"individual-response-how-answer": "Text message"})
+        self.post({"individual-response-enter-number-answer": "0763921456"})
```
Comment (JamesGardiner) [inline tests/integration/individual_response/test_individual_response.py:444]: Is this a synthetic number and has it been screened to check it doesn't belong to anyone?

**Review by JamesGardiner — COMMENTED** [review]
Couple of minor comments

**Review by MebinAbraham — COMMENTED** [review]
Happy with the changes, feature is as expected.
As discussed, just waiting on decision about lowering the max age for `URLSafeTimedSerializer` should be/whether there should be one and if not it should be `URLSafeSerializer`.

File: templates/individual_response/mobile_confirmation.html (hunk 2 above)
Comment (ajmaddaford) [reply templates/individual_response/mobile_confirmation.html:1]: I've gone with `confirmation-text-message.html` and renamed the other template to `confirmation-post.html`.

File: tests/integration/individual_response/test_individual_response.py (hunk 3 above)
Comment (ajmaddaford) [reply tests/integration/individual_response/test_individual_response.py:444]: I've just done a quick search and it doesn't look like it is. Ofcom lists appropriate ranges for dummy numbers - https://www.ofcom.org.uk/phones-telecoms-and-internet/information-for-industry/numbering/numbers-for-drama - so I'll create one in that range and update all of the tests.

File: app/views/handlers/individual_response.py [hunk 4]
```diff
@@ -622,3 +641,190 @@ def handle_post(self):
...
+                ),
+                "placeholders": IndividualResponseHandler._person_name_placeholder_possessive,
+            },
+            "answers": [
+                {
+                    "type": "TextField",
+                    "id": "individual-response-enter-number-answer",
+                    "mandatory": True,
+                    "label": "UK mobile number",
+                    "description": "This will not be stored and only used once to send the access code",
```
Comment (MebinAbraham) [inline app/views/handlers/individual_response.py:664]: Needs to be marked for translations.

File: app/views/handlers/individual_response.py [hunk 5]
```diff
@@ -622,3 +641,190 @@ def handle_post(self):
...
+        )
+
+    @staticmethod
+    def block_definition(mobile_number) -> Mapping:
+        return {
+            "type": "IndividualResponse",
+            "question": {
+                "type": "Question",
+                "id": "individual-response-text-confirm",
+                "title": "Is this mobile number correct?",
```
Comment (MebinAbraham) [inline app/views/handlers/individual_response.py:763]: Needs to be marked for translation.

**Review by MebinAbraham — APPROVED** [review]

File: tests/integration/individual_response/test_individual_response.py (hunk 3 above)
Comment (JamesGardiner) [reply tests/integration/individual_response/test_individual_response.py]: That's a good find - not seen it before but definitely good to know

**Review by JamesGardiner — APPROVED** [review]

**Pull request merged**